### PR TITLE
Android should use PointerDataPackets

### PR DESCRIPTION
**Pull request by abarth**
We now use JNI rather than Mojo to transport pointer data from Java to Dart.
Also, fill in a few more of the pointer data fields from information in Java.

### CODE REVIEW

**Comment by abarth** [comment]
@chinmaygarde @jamesr

File: shell/platform/android/io/flutter/view/FlutterView.java [hunk 1]
```diff
@@ -254,71 +252,118 @@ public InputConnection onCreateInputConnection(EditorInfo outAttrs) {
...
         }
         if (maskedAction == MotionEvent.ACTION_CANCEL) {
-            return PointerType.CANCEL;
+            return kPointerChangeCancel;
+        }
+        return -1;
+    }
+
+    private int getPointerDeviceTypeForToolType(int toolType) {
+        if (toolType == MotionEvent.TOOL_TYPE_FINGER) {
```
Comment (jamesr) [inline shell/platform/android/io/flutter/view/FlutterView.java]: switch?

File: shell/platform/android/io/flutter/view/FlutterView.java [hunk 2]
```diff
@@ -254,71 +252,118 @@ public InputConnection onCreateInputConnection(EditorInfo outAttrs) {
...
-        pointer.orientation = 0.0f;
-        pointer.tilt = 0.0f;
+        packet.putDouble(event.getAxisValue(MotionEvent.AXIS_ORIENTATION, pointerIndex)); // orientation
 
-        result.add(pointer);
+        if (pointerKind == kPointerDeviceKindStylus) {
+          packet.putDouble(event.getAxisValue(MotionEvent.AXIS_TILT, pointerIndex)); // tilt
+        } else {
+          packet.putDouble(0.0); // tilt
+        }
```
Comment (jamesr) [inline shell/platform/android/io/flutter/view/FlutterView.java]: wouldn't hurt to assert() here that this routine put the correct number of fields into |packet| to isolate busted serialization changes

**Comment by jamesr** [comment]
LGTM

**Pull request merged**